app/src/main/java/com/streamdata/apps/cryptochat/services/DateService.java

```diff
@@ -0,0 +1,41 @@
+package com.streamdata.apps.cryptochat.services;
+
+import android.app.Service;
+import android.content.Intent;
+import android.os.Binder;
+import android.os.IBinder;
+import android.support.annotation.Nullable;
+import android.util.Log;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Timer;
+import java.util.TimerTask;
+
+public class DateService extends Service{
+    private static Timer timer = new Timer();
+    private static final long REPEAT_TIME = 1000 * 30;
+
+    @Override
+    public int onStartCommand(Intent intent, int flags, int startId) {
+        timer.scheduleAtFixedRate(new Task(), 0, REPEAT_TIME);
+        return START_NOT_STICKY;
```

Comment:
For started services, there are two additional major modes of operation they can decide to run in, depending on the value they return from onStartCommand(): START_STICKY is used for services that are explicitly started and stopped as needed, while START_NOT_STICKY or START_REDELIVER_INTENT are used for services that should only remain running while processing any commands sent to them. See the linked documentation for more detail on the semantics.